### PR TITLE
fix: upgrade packages with vulns

### PR DESCRIPTION
**Pull request by guyb1**
## Description
Upgrade go deps with vulns

Fixed vulnerabilities in this PR:
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-24765' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30633' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-24675' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41771' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30631' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-28327' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30632' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41772' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-28131' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41103' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-27536' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-39293' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-29162' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-27664' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30630' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30635' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-30465' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-32189' },
 { pkg: 'git', cve: 'CVE-2022-31012' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-41716' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-32190' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-2879' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-2880' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-41720' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2019-2054' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-41715' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-1962' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-32148' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-31030' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-29526' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-32760' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-21334' },
 { pkg: '[golang.org/x/net](http://golang.org/x/net)', cve: 'CVE-2022-41717' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-24769' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-1705' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41089' },
 { pkg: '[github.com/moby/buildkit](http://github.com/moby/buildkit)', cve: 'CVE-2021-32760' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-34558' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-41717' },
 { pkg: '[github.com/containerd/containerd](http://github.com/containerd/containerd)', cve: 'CVE-2022-23471' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41091' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-36109' },
 { pkg: '[github.com/aws/aws-sdk-go](http://github.com/aws/aws-sdk-go)', cve: 'GHSA-f5pg-7wfw-84q9' },
 { pkg: '[github.com/aws/aws-sdk-go](http://github.com/aws/aws-sdk-go)', cve: 'GHSA-7f33-f4f5-xwgw' }

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -383,8 +384,7 @@ require (
 	sigs.k8s.io/yaml v1.3.0 // indirect
 )
 
-// See https://github.com/moby/moby/issues/42939#issuecomment-1114255529
```
Comment (knqyf263) [inline go.mod:386]: Did you remove the comment?

**Review by knqyf263 — COMMENTED** [review]
Could you explain which vulnerability we want to fix in the PR description?

File: go.mod (hunk 1 above)
Comment (guyb1) [reply go.mod:386]: added it back now

Comment (guyb1) [reply go.mod:386]: Done

File: go.mod [hunk 2]
```diff
@@ -384,7 +385,7 @@ require (
 )
 
 // See https://github.com/moby/moby/issues/42939#issuecomment-1114255529
-replace github.com/docker/docker => github.com/docker/docker v20.10.3-0.20220224222438-c78f6963a1c0+incompatible
+replace github.com/docker/docker => github.com/docker/engine v17.12.0-ce-rc1.0.20211209213653-8955d8da8951+incompatible
```
Comment (knqyf263) [inline go.mod:388]: Looks like it downgrades the versions. It is intended?

Comment (guyb1) [reply go.mod:388]: Its actually another package and it was auto-generated from its master branch.
Please take a look at this:
https://stackoverflow.com/questions/56032544/how-to-find-dependency-causing-sirupsen-logrus-vs-sirupsen-logrus-unexpecte

In the future we should consider upgrading to moby/moby.

Comment (knqyf263) [reply go.mod:388]: The commit you referenced is created on Dec 9, 2021. I don't think it addresses recent vulnerabilities. If scanners don't detect those vulnerabilities, it is an issue of scanners. Hiding vulnerabilities from scanners is not essential.
https://github.com/docker/engine/commit/8955d8da8951695a98eb7e15bead19d402c6eb27

**Comment by knqyf263** [comment]
Can you please review these vulnerabilities? Most of them don't seem relevant to Docker.

 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-24765' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30633' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-24675' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41771' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30631' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-28327' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30632' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41772' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-28131' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-41103' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-27536' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-39293' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-29162' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-27664' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30630' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-30635' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2021-30465' },
 { pkg: '[github.com/docker/docker](http://github.com/docker/docker)', cve: 'CVE-2022-32189' },

[CVE-2022-24765](https://nvd.nist.gov/vuln/detail/cve-2022-24765) is about Git for Windows. [CVE-2022-30633] is about Go, not Docker. I suspect these vulnerabilities are correct.

**Comment by knqyf263** [comment]
We bumped containerd.
https://github.com/aquasecurity/trivy/pull/3367

Please open PRs separately if you confirmed detected vulnerabilities.

**Pull request closed without merging**